packages/gamepads_web/lib/gamepads_web.dart

```diff
@@ -0,0 +1,137 @@
+// ignore_for_file: avoid_dynamic_calls, omit_local_variable_types
+
+import 'dart:async';
+import 'dart:js_interop';
+
+import 'package:flutter/foundation.dart';
+import 'package:flutter_web_plugins/flutter_web_plugins.dart';
+import 'package:gamepads_platform_interface/api/gamepad_controller.dart';
+import 'package:gamepads_platform_interface/api/gamepad_event.dart';
+import 'package:gamepads_platform_interface/gamepads_platform_interface.dart';
+
+import 'package:gamepads_web/gamepad_detector.dart';
+import 'package:web/web.dart' as web;
+
+class GamePadState {
+  GamePadState(int length) {
+    keyStates = List<dynamic>.filled(length, null);
+    axesStates = List<dynamic>.filled(4, null);
+  }
+
+  List<dynamic>? keyStates;
+  List<dynamic>? axesStates;
+}
+
+/// A web implementation of the GamepadsWebPlatform of the GamepadsWeb plugin.
+class GamepadsWeb extends GamepadsPlatformInterface {
+  int _gamepadCount = 0;
+  Timer? _gamepadPollingTimer;
+
+  Map<String, GamePadState> lastGamePadstates = {};
+
+  void updateGamepadsStatus() {
+    final gamepads = getGamepadList();
+    for (int i = 0; i < gamepads.length; i++) {
+      final gamepad = gamepads[i];
+      if (gamepad != null) {
```

Comment:
Use for-in instead, and since gamepad can't be null coming from `getGamepadList` this check isn't needed (that's also why `getGamepadList` needs a stronger return type.